### PR TITLE
Adding Openai v1.0.0+ support into data index

### PR DESCRIPTION
**Pull request by shpengmsft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by luigiw — APPROVED** [review]

File: sdk/ai/azure-ai-resources/setup.py [hunk 1]
```diff
@@ -70,6 +70,7 @@
         "mlflow-skinny<3",
         "azureml-telemetry~=1.0,>=1.51.0",
         "azure-mgmt-resource<23.0.0,>=22.0.0",
+        "tqdm",
```
Comment (luigiw) [inline sdk/ai/azure-ai-resources/setup.py:73]: Where is this used?

Comment (shpengmsft) [reply sdk/ai/azure-ai-resources/setup.py:73]: https://github.com/Azure/azure-sdk-for-python/blob/ad5ddf4e71341e51ce09b36342bf52fe73a555b3/sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py#L204

Comment (luigiw) [reply sdk/ai/azure-ai-resources/setup.py:73]: I see. It looks like an optional dependency. Let's keep it out of the install_requires list to minimize the dependencies. The AI SDK now suffers the problem of too many dependencies and it takes up to 20 mins to install both the ai-resources and ai-generative packages. I want to be extra careful about adding new dependencies.

Comment (shpengmsft) [reply sdk/ai/azure-ai-resources/setup.py:73]: In fact, I added it there because I was trying to fix followng build error -

**Review by luigiw — APPROVED** [review]

**Review by jinzhaochangAtMicrosoft — APPROVED** [review]

**Pull request merged**